### PR TITLE
chore: [app-router-migration-4]: apps/categories page

### PR DESCRIPTION
**Pull request by dmytro068**
## What does this PR do?

Migrates apps/categories, apps/categories/[category] pages.

This PR depends on https://github.com/calcom/cal.com/pull/12593

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Chore (refactoring code, technical debt, workflow improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

**Review by hariombalhara — DISMISSED** [review]
I don't know the reason but I am seeing a consistent crash on accessing http://app.cal.local:3000/apps/categories with future mode.
https://www.loom.com/share/66d37da1d31940c39691f04d7e695e0a

**Comment by DmytroHryshyn** [comment]
> I don't know the reason but I am seeing a consistent crash on accessing http://app.cal.local:3000/apps/categories with future mode. https://www.loom.com/share/66d37da1d31940c39691f04d7e695e0a

This issue can be related to https://github.com/vercel/next.js/issues/54708. App dir router memory consumption is significantly higher then for Pages dir, but this does not affect production build, this is only dev server problem. It can be temporarily fixed by adding `NODE_OPTIONS="--max-old-space-size=8192"` before `yarn dev` command. We are researching if there is better fix for this.

File: apps/web/abTest/middlewareFactory.ts [hunk 1]
```diff
@@ -6,6 +6,9 @@ import z from "zod";
 const ROUTES: [URLPattern, boolean][] = [
   ["/event-types", process.env.APP_ROUTER_EVENT_TYPES_ENABLED === "1"] as const,
   ["/settings/admin/:path*", process.env.APP_ROUTER_SETTINGS_ADMIN_ENABLED === "1"] as const,
+  ["/apps/categories", Boolean(process.env.APP_ROUTER_APPS_CATEGORIES_ENABLED)] as const,
```
[the remark this answers is not in the document]
Comment (DmytroHryshyn) [reply apps/web/abTest/middlewareFactory.ts]: Thanks for pointing this out, this was a problem in our internal codemod that generates url patterns and flags. Already fixed this.

**Review by zomars — APPROVED** [review]
LGTM

**Pull request merged**